vector/src/main/java/im/vector/app/features/location/live/StopLiveLocationShareUseCase.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.location.live
+
+import org.matrix.android.sdk.api.session.Session
+import org.matrix.android.sdk.api.session.getRoom
+import org.matrix.android.sdk.api.session.room.location.UpdateLiveLocationShareResult
+import javax.inject.Inject
+
+class StopLiveLocationShareUseCase @Inject constructor(
+        private val session: Session
```

Comment:
I can't remember the outcome of this, if the `StopLiveLocationShareUseCase` instance can exist across sessions, the inject session will be invalid

might be safer to inject and lazily fetch from the `ActiveSessionHolder`, or scope the `UseCase` to session?

Reply:
You are right. Find a fix here: [9450267](https://github.com/vector-im/element-android/pull/6356/commits/945026730c8cb29d26f86e35b5d260ea03b70168)